dist/bootstrap/index.js

```diff
@@ -5674,7 +5674,7 @@ function run() {
         const juju_channel = core.getInput("juju-channel");
         const juju_bundle_channel = core.getInput("juju-bundle-channel");
         const juju_crashdump_channel = core.getInput("juju-crashdump-channel");
-        const lxd_channel = core.getInput("lxd-channel");
+        const lxd_channel = (provider === "lxd" && ![null, ""].includes(channel)) ? channel : core.getInput("lxd-channel");
```

Comment:
If my TS/JS lenses are still working, you can use channel directly. `null` and `""` are both falsy values.
```suggestion
        const lxd_channel = (provider === "lxd" && channel ? channel : core.getInput("lxd-channel");
```